### PR TITLE
OASIS brain StyleGAN

### PR DESCRIPTION
**Pull request by EthanGTJones**
I have attempted to create a StyleGAN model to generate synthetic MRI brain scans using the OASIS brain dataset.

After training the model with a batch size of 32 with 100 epochs, generator learning rate of 0.00001 and a discriminator learning rate of  0.0000125, I believe the results satisfy the primary condition of producing reasonably clear images. The generator and discriminator loss converged at around 5 epochs, however at around 50 epochs, there was a significant model collapse. 

Given the required training time for the model config above, I was unable to adjust the model parameters and retrain the model. If I had the time to rerun the model, I would decrease the learning rate for both generator and the discriminator.

### CODE REVIEW

**Comment by WeiDaiwow** [comment]
# Document
 
- Title: 
	Good
- Address problem:
	Problem mentioned, but not clearly addressed. Why using GAN instead of other models for this particular task.
- Working Principle:
	Good
- Figure visualization:
	Good
- dependency:
	Good
- inputs and outputs description:
	No input visualization.
        Output visualization provided.
- training plot:
	Good
- readme using markdown:
	Good
- preprocessing description:
	Good
- Description of usage:
	very limited description. not very clear on how to use each file.



# Check the files:
- modules.py
	Good
- dataset.py
	Good
- train.py
	Good
- predict.py
	Good
Suggestion:
1. during testing stage, you should disable the training mode of generated when generating images. generator(images, training = False)


# Overall:

- solve the problem:
	Yes
- TF/torch implementation:
	Yes
- Design quality:
	Only the AdaIN layer was implemented, other techniques such as progressive training, gradient penalty etc. were not 
        implemented. 
         The fine details in generated images are not very good
- code comments:
	Good
- Difficulty: 
	StyleGANv1 - hard

**Comment by EthanGTJones** [comment]
Morning Wei, 

Thank you for your feedback! I have added your recommended changes to my README file. 

As for the results, I am not sure that I will have time to allow the model to run again and produce higher quality images. My choice of hyperparameters were primarily due to time constraints, But I would definitely like to increase the EPOCH count to see some better images.  

Thanks again and enjoy your summer!

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
